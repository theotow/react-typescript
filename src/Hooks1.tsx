import React, { useReducer } from 'react';
import immer from 'immer';

interface State {
  count: number;
}

interface Action {
  type: string;
  whatever: string;
}

function reducer(state: State, action: Action) {
  switch (action.type) {
    case 'increase':
      return immer(state, (snap) => {
        snap.count += 1;
      });
    default:
      return state;
  }
}

export default function Hooks1() {
  const [state, dispatch] = useReducer(reducer, { count: 0 });
  return (
    <div>
      <p>You clicked {state.count} times</p>
      // @ts-ignore
      <button onClick={() => dispatch({ type: 'increase' })}>Click me</button>
    </div>
  );
}
