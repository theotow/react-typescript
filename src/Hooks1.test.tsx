import Hooks1 from './Hooks1';
import React from 'react';
import { mount } from 'enzyme';

test('hooks', () => {
  const ele = mount(<Hooks1 />);
  ele.find('button').simulate('click');
  expect(ele).toMatchSnapshot();
});
