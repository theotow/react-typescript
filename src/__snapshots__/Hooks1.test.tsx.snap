// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks 1`] = `
<Hooks1>
  <div>
    <p>
      You clicked 
      1
       times
    </p>
    <button
      onClick={[Function]}
    >
      Click me
    </button>
  </div>
</Hooks1>
`;
